### PR TITLE
Rename language codes (Icelandic, multi-language)

### PR DESCRIPTION
**Pull request by thomashacker**
## Description

Renames the Icelandic language code from `is` to `isl` and the multi-language code `xx` to `mul`

### Types of change

Enhancement

## Checklist

- [x] I confirm that I have the right to submit this contribution under the project's MIT license.
- [x] I ran the tests, and all new and existing tests passed.
- [x] My changes don't require a change to the documentation, or if they do, I've added all required information.

### CODE REVIEW

File: spacy/errors.py [hunk 1]
```diff
@@ -962,6 +962,15 @@ class Errors(metaclass=ErrorsWithCodes):
...
     E4004 = ("Backprop is not supported when is_train is not set.")
 
+# Deprecated model shortcuts, only used in errors and warnings
+OLD_MODEL_SHORTCUTS = {
+    "en": "en_core_web_sm", "de": "de_core_news_sm", "es": "es_core_news_sm",
+    "pt": "pt_core_news_sm", "fr": "fr_core_news_sm", "it": "it_core_news_sm",
+    "nl": "nl_core_news_sm", "el": "el_core_news_sm", "nb": "nb_core_news_sm",
+    "lt": "lt_core_news_sm", "xx": "xx_ent_wiki_sm"
+}
+
```
Comment (adrianeboyd) [inline spacy/errors.py:972]: This should stay removed:

```suggestion
```

File: spacy/cli/_util.py [hunk 2]
```diff
@@ -131,6 +132,14 @@ def _parse_override(value: Any) -> Any:
     except ValueError:
         return str(value)
 
+def _handle_renamed_language_codes(lang: Optional[str]) -> None:
+    # Throw error for renamed language codes in v4
+    if lang in RENAMED_LANGUAGE_CODES:
+        msg.fail(
+            title="Renamed language code",
+            text=f"Language code '{lang}' was replaced with '{RENAMED_LANGUAGE_CODES[lang]}' in spaCy v4. Update the language code from '{lang}' to '{RENAMED_LANGUAGE_CODES[lang]}'.",
```
Comment (svlandeg) [inline spacy/cli/_util.py:140]: Can you make sure to run `black` on all the code files?

File: website/meta/languages.json [hunk 3]
```diff
@@ -434,7 +434,7 @@
             ]
         },
         {
-            "code": "xx",
+            "code": "mul",
             "name": "Multi-language",
             "models": ["xx_ent_wiki_sm", "xx_sent_ud_sm"],
```
Comment (svlandeg) [inline website/meta/languages.json:439]: I assume we'll rename these models too, right?

**Review by svlandeg — COMMENTED** [review]
I also found a few more occurrences of `xx`:
* `spacy\tests\README.md`
* `scorer.mdx`
* `quickstart-models.js`

File: website/meta/languages.json (hunk 3 above)
Comment (thomashacker) [reply website/meta/languages.json:439]: So many "xx"! 😄 I'll go through the codebase again and rename all remaining "xx"

File: spacy/cli/_util.py (hunk 2 above)
Comment (thomashacker) [reply spacy/cli/_util.py:140]: Weird, isn't this `black` formatted? I double-checked, and it always formats to this. Maybe something is configured wrong on my end?

Comment (svlandeg) [reply spacy/cli/_util.py:140]: I'm pretty sure it isn't correctly formatted, no, as there should be 2 blank lines (not 1) before and after the function?

Comment (thomashacker) [reply spacy/cli/_util.py:141]: Ah yes, you're right! Pushed the change and made sure that `black` is run on all the other 29 files. I wasn't paying attention to the trailing spaces but only looked at the code block, sorry!

File: website/docs/api/scorer.mdx [hunk 4]
```diff
@@ -30,7 +30,7 @@ Create a new `Scorer`.
...
 | ------------------ | ----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
 | `nlp`              | The pipeline to use for scoring, where each pipeline component may provide a scoring method. If none is provided, then a default pipeline is constructed using the `default_lang` and `default_pipeline` settings. ~~Optional[Language]~~ |
-| `default_lang`     | The language to use for a default pipeline if `nlp` is not provided. Defaults to `xx`. ~~str~~                                                                                                                                            |
+| `default_lang`     | The language to use for a default pipeline if `nlp` is not provided. Defaults to `mul`. ~~str~~                                                                                                                                            |
```
Comment (svlandeg) [inline website/docs/api/scorer.mdx:33]: Almost there - just another final round of `prettier` will be required for the changes to the `md(x)` files ;-)

**Pull request merged**